SearchableSnapshotsLicenseIntegTests.testShardAllocationOnInvalidLicense fails when recreating its license

<!--
Please fill out the following information, and ensure you have attempted
to reproduce locally
-->

**Build scan**:
https://gradle-enterprise.elastic.co/s/2bpdnr5jqux6y/tests

**Repro line**:
```
./gradlew ':x-pack:plugin:searchable-snapshots:internalClusterTest' --tests "org.elasticsearch.xpack.searchablesnapshots.SearchableSnapshotsLicenseIntegTests.testShardAllocationOnInvalidLicense" -Dtests.seed=5470D1E16498B727 -Dtests.locale=be -Dtests.timezone=Canada/Pacific -Druntime.java=11
```
**Reproduces locally?**: no

**Applicable branches**: master

**Failure history**:
https://build-stats.elastic.co/goto/d7017643bd9139fb127d1f231d6d81d2

![image](https://user-images.githubusercontent.com/215970/116301902-900dd980-a76e-11eb-9b99-ec97b86c0ee3.png)

**Failure excerpt**:

```
java.lang.AssertionError: expected:<UPGRADED_TO_TRIAL> but was:<TRIAL_ALREADY_ACTIVATED> |  
...
at __randomizedtesting.SeedInfo.seed([5470D1E16498B727:D8600A79DC686337]:0)
```

--- a/x-pack/plugin/searchable-snapshots/src/internalClusterTest/java/org/elasticsearch/xpack/searchablesnapshots/SearchableSnapshotsLicenseIntegTests.java
+++ b/x-pack/plugin/searchable-snapshots/src/internalClusterTest/java/org/elasticsearch/xpack/searchablesnapshots/SearchableSnapshotsLicenseIntegTests.java
@@ -89,6 +89,9 @@ public class SearchableSnapshotsLicenseIntegTests extends BaseFrozenSearchableSn
 
         assertAcked(client().execute(DeleteLicenseAction.INSTANCE, new DeleteLicenseRequest()).get());
         assertAcked(client().execute(PostStartBasicAction.INSTANCE, new PostStartBasicRequest()).get());
+
+        ensureClusterSizeConsistency();
+        ensureClusterStateConsistency();
     }
 
     public void testMountRequiresLicense() {
@@ -150,6 +153,10 @@ public class SearchableSnapshotsLicenseIntegTests extends BaseFrozenSearchableSn
                 client().admin().cluster().prepareHealth(indexName).get().getIndices().get(indexName).getStatus()
             )
         );
+
+        waitNoPendingTasksOnAll();
+        ensureClusterStateConsistency();
+
         // add a valid license again
         // This is a bit of a hack in tests, as we can't readd a trial license
         // We force this by clearing the existing basic license first
@@ -158,6 +165,10 @@ public class SearchableSnapshotsLicenseIntegTests extends BaseFrozenSearchableSn
                 .metadata(Metadata.builder(currentState.metadata()).removeCustom(LicensesMetadata.TYPE).build())
                 .build()
         );
+
+        waitNoPendingTasksOnAll();
+        ensureClusterStateConsistency();
+
         PostStartTrialRequest startTrialRequest = new PostStartTrialRequest().setType(License.LicenseType.TRIAL.getTypeName())
             .acknowledge(true);
         PostStartTrialResponse resp = client().execute(PostStartTrialAction.INSTANCE, startTrialRequest).get();

